core/src/test/java/org/apache/lucene/queries/LongRandomBinaryDocValuesRangeQueryTests.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.lucene.queries;
+
+import org.elasticsearch.index.mapper.RangeFieldMapper;
+
+public class LongRandomBinaryDocValuesRangeQueryTests extends BaseRandomBinaryDocValuesRangeQueryTestCase {
+
+    @Override
+    protected String fieldName() {
+        return "long_range_dv_field";
+    }
+
+    @Override
+    protected RangeFieldMapper.RangeType rangeType() {
+        return RangeFieldMapper.RangeType.LONG;
+    }
+
+    @Override
+    protected Range nextRange(int dimensions) throws Exception {
+        long value1 = nextLongInternal();
+        long value2 = nextLongInternal();
+        long min = Math.min(value1, value2);
+        long max = Math.max(value1, value2);
+        return new LongTestRange(min, max);
+    }
+
+    private long nextLongInternal() {
+        if (rarely()) {
+            return random().nextBoolean() ? Long.MAX_VALUE : Long.MIN_VALUE;
+        }
+        long max = Long.MAX_VALUE / 2;
+        return (max + max) * random().nextLong() - max;
+    }
```

Comment:
We had similar random number generation in eg. `TestIntRangeFieldQueries` but this proved inefficient at finding bugs (eg. it makes it very unlikely to get twice the same value in the same test) so we changed it in https://issues.apache.org/jira/browse/LUCENE-7847. Maybe we should do something similar here (as well as for doubles, ip addresses, etc.)